tests/common/params_test.py

```diff
@@ -31,11 +31,22 @@ def test_overrides(self):
 
     def test_as_flat_dict(self):
         filename = 'tests/fixtures/bidaf/experiment.json'
-        params = Params.from_file(filename).as_flat_dict()
+        params = Params({
+            'a': 10,
+            'b': {
+                'c': 20,
+                'd': 'stuff'
+            }
+        }).as_flat_dict()
+
+        assert "a" in params
+        assert params["a"] == 10
+
+        assert "b.c" in params
+        assert params["b.c"] == 20
 
-        assert "dataset_reader.type" in params
-        assert "dataset_reader.token_indexers.tokens.lowercase_tokens" in params
-        assert "model.text_field_embedder.tokens.type" in params
+        assert "b.d" in params
+        assert params["b.d"] == 'stuff'
```

Comment:
This test still doesn't quite do what I suggested.  You really should just assert dictionary equality:

```python
assert params == {'a': 10, 'b.c': 20, 'b.d': 'stuff'}
```

Reply:
sigh--you're right.  Done.